src/main/webapp/app/more-feedback-for-tutor/more-feedback-for-tutor.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { JhiAlertService } from 'ng-jhipster';
+import { ComplaintService } from 'app/entities/complaint/complaint.service';
+import { HttpErrorResponse } from '@angular/common/http';
+import { ComplaintResponseService } from 'app/entities/complaint-response/complaint-response.service';
+import { ComplaintResponse } from 'app/entities/complaint-response';
+import { Complaint, ComplaintType } from 'app/entities/complaint';
+
+@Component({
+    selector: 'jhi-more-feedback-for-tutor-form',
+    templateUrl: './more-feedback-for-tutor.component.html',
+    providers: [JhiAlertService],
+})
+export class MoreFeedbackForTutorComponent implements OnInit {
+    @Input() complaint: Complaint;
+    @Input() isAllowedToRespond: boolean; // indicates if the tutor is allowed to respond
+    complaintText = '';
+    handled: boolean;
+    complaintResponse: ComplaintResponse = new ComplaintResponse();
+
+    constructor(private complaintService: ComplaintService, private jhiAlertService: JhiAlertService, private complaintResponseService: ComplaintResponseService) {}
+
+    ngOnInit(): void {
+        this.complaintText = this.complaint.complaintText;
+        this.handled = this.complaint.accepted !== undefined;
+        if (this.handled) {
+            this.complaintResponseService.findByComplaintId(this.complaint.id).subscribe(complaintResponse => (this.complaintResponse = complaintResponse.body!));
+        }
+    }
+
+    respondToComplaint(): void {
+        if (this.complaintResponse.responseText.length <= 0 || !this.isAllowedToRespond) {
+            return;
+        }
+        this.handled = true;
```

Comment:
This is an optimistic approach to handle API requests.
But what if it fails? Then the complaint in the UI is still accepted.

I think it's a better approach to show a loading indicator while the server is processing the request and then update the complaint to be handled / accepted.